src/AnnotationThread.js

```diff
@@ -113,94 +130,148 @@ class AnnotationThread extends EventEmitter {
+        this.updateTemporaryAnnotation = this.updateTemporaryAnnotation.bind(this);
+        this.delete = this.delete.bind(this);
+
+        const pageEl = this.getPopoverParent();
+        this.popoverComponent = render(
+            <AnnotationPopover
+                id={this.id}
+                type={this.type}
+                isMobile={util.shouldDisplayMobileUI(this.container)}
+                createdAt={this.createdAt}
+                createdBy={this.createdBy}
+                modifiedAt={this.modifiedAt}
+                canAnnotate={this.canAnnotate}
+                canDelete={this.canDelete}
+                canComment={this.canComment}
+                comments={this.comments}
+                position={this.position}
+                onDelete={this.delete}
+                onCancel={this.cancelUnsavedAnnotation}
+                onCreate={this.save}
+                onCommentClick={this.onCommentClick}
+                isPending={isPending}
+                headerHeight={this.headerHeight}
+            />,
+            util.getPopoverLayer(pageEl)
+        );
     }
 
-    /**
-     * Hides the appropriate annotation dialog for this thread.
-     *
-     * @return {void}
-     */
-    hideDialog() {
-        if (this.dialog) {
-            this.state = STATES.inactive;
-            this.dialog.hide();
+    unmountPopover() {
+        this.reset();
+
+        this.toggleFlippedThreadEl();
+
+        const pageEl = this.getPopoverParent();
+        const popoverLayer = pageEl.querySelector('.ba-dialog-layer');
+        if (this.popoverComponent && popoverLayer) {
+            unmountComponentAtNode(popoverLayer);
+            this.popoverComponent = null;
         }
     }
 
     /**
      * Saves an annotation.
      *
      * @param {string} type - Type of annotation
-     * @param {string} text - Text of annotation to save
+     * @param {string} message - Text of annotation to save
      * @return {Promise} - Annotation create promise
      */
-    saveAnnotation(type, text) {
-        const annotationData = this.createAnnotationData(type, text);
+    save(type, message) {
+        const annotationData = this.createAnnotationData(type, message);
 
         // Save annotation on client
-        const tempAnnotationID = AnnotationService.generateID();
-        const tempAnnotationData = annotationData;
-        tempAnnotationData.annotationID = tempAnnotationID;
-        tempAnnotationData.permissions = {
-            can_edit: true,
-            can_delete: true
-        };
-        tempAnnotationData.created = new Date().getTime();
-        tempAnnotationData.modified = tempAnnotationData.created;
-        const tempAnnotation = new Annotation(tempAnnotationData);
-        this.saveAnnotationToThread(tempAnnotation);
-        this.state = STATES.inactive;
+        const id = AnnotationAPI.generateID();
+        this.updateAnnotationThread({
+            id,
+            message,
+            permissions: {
```

Comment:
Should this contain the permissions provided from the API, not just be hardcoded to true?